core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala

```diff
@@ -87,19 +110,31 @@ class ExecutorsListener(storageStatusListener: StorageStatusListener, conf: Spar
         s.blockManagerId.executorId == SparkContext.LEGACY_DRIVER_IDENTIFIER ||
         s.blockManagerId.executorId == SparkContext.DRIVER_IDENTIFIER
       }
-      storageStatus.foreach { s => executorToLogUrls(s.blockManagerId.executorId) = logs.toMap }
+      storageStatus.foreach { s =>
+        val eid = s.blockManagerId.executorId
+        if (!executorToTaskSummary.contains(eid)) {
+          executorToTaskSummary(eid) = ExecutorTaskSummary(eid)
+        }
+        executorToTaskSummary(eid).executorLogs = logs.toMap
+      }
     }
   }
 
   override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = synchronized {
     val eid = taskStart.taskInfo.executorId
-    executorToTasksActive(eid) = executorToTasksActive.getOrElse(eid, 0) + 1
+    if (!executorToTaskSummary.contains(eid)) {
+      executorToTaskSummary(eid) = ExecutorTaskSummary(eid)
+    }
+    executorToTaskSummary(eid).tasksActive += 1
   }
 
   override def onTaskEnd(taskEnd: SparkListenerTaskEnd): Unit = synchronized {
     val info = taskEnd.taskInfo
     if (info != null) {
       val eid = info.executorId
+      if (!executorToTaskSummary.contains(eid)) {
```

Comment:
Same comment around this block, seems like executorToTaskSummary(eid) can be retrieved once and simplify this a bit.